lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +50,45 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_f = list(self.path_to_raw_txt_data.glob("*_raw.txt"))
+        meta_f = list(self.path_to_raw_txt_data.glob("*_meta.json"))
+        if len(meta_f) != len(raw_f):
+            raise InconsistentDatasetError
+        sorted_raw_files = sorted(raw_f)
+        sorted_meta_files = sorted(meta_f)
+
+        for ind, (raw, meta) in enumerate(zip(sorted_raw_files, sorted_meta_files)):
+            if ind + 1 != get_article_id_from_filepath(raw) \
+                    or ind + 1 != get_article_id_from_filepath(meta):
+                raise InconsistentDatasetError
+
+        if any(file.stat().st_size == 0 for file in (raw_f + meta_f)):
+            raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        for file in list(self.path_to_raw_txt_data.glob("*_raw.txt")):
+            art_id = get_article_id_from_filepath(file)
+            self._storage[art_id] = from_raw(file, Article(None, art_id))
```

Comment:
Is it necessary to specify the second argument?

Reply:
no, actually it can be removed because this argument is optional and equals None if we do not specify it. so the function works without it